### ISSUE TITLE
Settingtype.txt line ending

### ISSUE THREAD

**Issue by sfence** [minetest/minetest#11327]
##### Minetest version 
```
Minetest 5.4.1 (Linux)
Using Irrlicht 1.8.4
Using LuaJIT 2.0.5
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
USE_FREETYPE=1
STATIC_SHAREDIR="/usr/share/minetest"
STATIC_LOCALEDIR="/usr/share/locale"
```

##### OS
Arch linux

##### Summary
Biofuel mod settingtypes.txt file cause the problem:
```
2021-06-09 08:18:04: ERROR[Main]: Invalid integer setting in /home/sfence/.minetest/mods/biofuel/settingtypes.txt "fuel_production_time (fuel production time) i"t 10 5 360
2021-06-09 08:18:04: ERROR[Main]: Invalid integer setting in /home/sfence/.minet"st/mods/biofuel/settingtypes.txt "biomass_input (required input) int 4 1 99
2021-06-09 08:18:04: ERROR[Main]: Invalid boolean setting in /home/sfence/.minet"st/mods/biofuel/settingtypes.txt "refinery_output (bottle output) bool false
2021-06-09 08:18:04: ERROR[Main]: Invalid boolean setting in /home/sfence/.minet"st/mods/biofuel/settingtypes.txt "food_fuel (food waste) bool false
```
Problem disappears when the line ends from settingtypes.txt is changed from dos style to unix style.

##### Steps to reproduce
Use biofuel mod https://github.com/Lokrates/Biofuel/commit/6ad39ec55ed0def48167e91f31381daaf31ee9ba 
Maybe this bug is platform is specific.

**Comment by neoh4x0r** [comment]
We need to strip carriage returns (CR, \r, 0x0d) from the line parameter in builtin/mainmenu/dlg_settings_advanced.lua
https://github.com/minetest/minetest/blob/master/builtin/mainmenu/dlg_settings_advanced.lua#L33

For example it could be done like this
```
-- returns error message, or nil
local function parse_setting_line(settings, line, read_all, base_level, allow_secure, b_biofuel)

	-- strip carriage returns (CR, /r)
	line = line:gsub("\r","")

	-- comment
	local comment = line:match("^#" .. CHAR_CLASSES.SPACE .. "*(.*)$")
```

**Comment by neoh4x0r** [comment]
@sfence Can you test the fix in PR #11338 that was just merged today ? That should fix this issue.

**Comment by sfence** [comment]
@neoh4x0r It looks like this error disappears in minetest build from the actual master branch. So it works well probably.
